apps/examples/testcase/le_tc/kernel/itc_semaphore.c

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_semaphore.c
+/// @brief Test Case Example for semaphore API
+#include <tinyara/config.h>
+#include <semaphore.h>
+#include <pthread.h>
+#include "tc_internal.h"
+#include <stdio.h>
+#include <sys/types.h>
+
+#define LOOP_SIZE CONFIG_ITC_KERN_SEM_LOOP_SIZE
+#define PSHARED 0
+#define SEM_COUNT 1
+
+/**
+* @fn                   :itc_semaphore_sem_init_destroy_p_multitime
+* @brief                :sem_destroy destroys semaphore
+* @scenario             :sem init destroy multitime
+* API's covered         :sem_init,sem_destroy
+* Preconditions         :none
+* Postconditions        :none
+*/
+static void itc_semaphore_sem_init_destroy_p_multitime(void)
+{
+	sem_t sem;
+	int ret_chk;
+	int exec_index;
+
+	for (exec_index = 0; exec_index < LOOP_SIZE; exec_index++) {
+		ret_chk = sem_init(&sem, PSHARED, SEM_COUNT);
+		TC_ASSERT_EQ("sem_init", ret_chk, OK);
+
+		ret_chk = sem_destroy(&sem);
+		TC_ASSERT_EQ("sem_destroy", ret_chk, OK);
+		TC_ASSERT_EQ("sem_destroy", sem.semcount, SEM_COUNT);
```

Comment:
sem_destroy will change sem.semcount to 1 when semcount is greater than 0.
So I think you'd better set initial semcount to another number for checking destroyed well.

Reply:
As per sem_destroy manual whcih descrived below
sem_destroy() destroys the unnamed semaphore at the address pointed
       to by sem.

       Only a semaphore that has been initialized by sem_init(3) should be
       destroyed using sem_destroy().

       Destroying a semaphore that other processes or threads are currently
       blocked on (in sem_wait(3)) produces undefined behavior.

       Using a semaphore that has been destroyed produces undefined results,
       until the semaphore has been reinitialized using sem_init(3).
So I am not getting,  what you mean by your statement. Please explain further

Reply:
In TizenRT, when called sem_destroy , it will set semcount to 1.(not really destroy the semaphore, only set the count to 1)
So I think, if you call sem_init with greater than 1(for example, sem_init(&sem, PSHARED, 3), semcount will be set to 3.
And if you destroy that sem, then semcount will be 1.
After destroying, if semcount is not 1, then error. if 1, then pass.

Reply:
As per the comments,the required changes has been made in the itc semaphore testcase.